### PR TITLE
Representative Image: Enable also in pages

### PR DESCRIPTION
**Pull request by pauloxnet**
- Enable representative image plugin to work in `Page`
- Add `requirements.txt`
- Update .`gitignore`

### CODE REVIEW

File: representative_image/test_representative_image.py [hunk 1]
```diff
@@ -56,6 +56,18 @@ def test_extract_image_from_summary_with_custom_image(self):
         self.assertEqual(article.featured_image, TEST_CUSTOM_IMAGE_URL)
         self.assertEqual(article.summary, TEST_SUMMARY_WITHOUTIMAGE)
 
+    def test_extract_image_from_page_summary_with_custom_image(self):
```
Comment (Lucas-C) [inline representative_image/test_representative_image.py]: 👍

File: representative_image/Readme.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-The plugin also removes any images from the summary after extraction to avoid duplication. 
+The plugin also removes any images from the summary after extraction to avoid duplication.
 
-It allows the flexibility on where and how to display the featured image of an article together with its summary in a template page. For example, the article metadata can be displayed in thumbnail format, in which there is a short summary and an image. The layout of the summary and the image can be varied for aesthetical purpose. It doesn't have to depend on article's content format. 
+It allows the flexibility on where and how to display the featured image of an article together with its summary in a template page. For example, the article metadata can be displayed in thumbnail format, in which there is a short summary and an image. The layout of the summary and the image can be varied for aesthetical purpose. It doesn't have to depend on article's content format.
 
 ## Installation
 
-This plugin requires BeautifulSoup:
+This plugin requires `BeautifulSoup`:
```
Comment (Lucas-C) [inline representative_image/Readme.md]: Could you use this opportunity to a `requirements.txt` in this plugin directory please ?

Comment (pauloxnet) [reply representative_image/Readme.md]: Added in https://github.com/getpelican/pelican-plugins/pull/1220/commits/0d71bdb636711a62ce2058f3a86ef5e085034e8b

File: .gitignore [hunk 3]
```diff
@@ -9,3 +9,5 @@
 .idea
 .project
 .pydevproject
+__pycache__
```
Comment (Lucas-C) [inline .gitignore]: 👍

**Comment by Lucas-C** [comment]
Good for me !

**Comment by justinmayer** [comment]
Many thanks to Paolo for the enhancement and to Lucas for reviewing! ✨

**Comment by justinmayer** [comment]
Hmm. Looks like Travis is [reporting test failures](https://travis-ci.org/getpelican/pelican-plugins/builds/599322436). Any thoughts on that, @pauloxnet / @Lucas-C ?

**Comment by pauloxnet** [comment]
@justinmayer I'm giving a check on it. It passed when I pushed my latest commit
https://travis-ci.org/getpelican/pelican-plugins/builds/592987041

**Comment by pauloxnet** [comment]
@justinmayer I checked the tests failed in a plugin different from the one (`representative_image`) I modified in my PR, as you can see in the failed tests logs:
```python
...
Traceback (most recent call last):
  File "/home/travis/build/getpelican/pelican-plugins/i18n_subsites/test_i18n_subsites.py", line 138, in test_sites_generation
    self.assertFalse(out, 'non-empty `diff` stdout:\n{}'.format(out))
...
```
I can't see in travis the button to trigger a new build, can you do that so we can check which test are failing ?

**Comment by justinmayer** [comment]
Triggered another build, with same result: `FAIL: Test generation of sites with the plugin`

**Comment by avaris** [comment]
From what I can see, error is related to [this update](https://github.com/getpelican/pelican/commit/f72d06a4a5271d1a4156ef042f307a479dd7b452) from `pelican`. `i18n_subsites` test output requires regeneration.

**Comment by pauloxnet** [comment]
> From what I can see, error is related to [this update](https://github.com/getpelican/pelican/commit/f72d06a4a5271d1a4156ef042f307a479dd7b452) from `pelican`. `i18n_subsites` test output requires regeneration.

Thanks for the review.

@justinmayer we have to open an issue related to `i18n_subsites` plugin tests ?

**Comment by Lucas-C** [comment]
I opened a PR to fix those tests: https://github.com/getpelican/pelican-plugins/pull/1225

**Pull request merged**